src/test/java/org/folio/rest/api/HridSettingsStorageTest.java

```diff
@@ -478,6 +608,29 @@ public void canGetNextHridWhenStartNumberIsLong(TestContext testContext) {
       .onComplete(testContext.asyncAssertSuccess());
   }
 
+  @Test
+  public void canGetNextHridWhenStartNumberIsLongWithoutLeadingZeroes(TestContext testContext) {
+    log.info("Starting canGetNextHridWhenStartNumberIsLongWithoutLeadingZeroes()");
+    final Vertx vertx = StorageTestSuite.getVertx();
+    final PostgresClient postgresClient = PostgresClient.getInstance(vertx, TENANT_ID);
+
+    final HridManager hridManager = new HridManager(vertx.getOrCreateContext(), postgresClient);
+    final HridSettings newHridSettings = new HridSettings()
+      .withInstances(new HridSetting().withStartNumber(9_999_999_997L).withRetainLeadingZeroes(false))
+      .withHoldings(new HridSetting().withStartNumber(9_999_999_998L).withRetainLeadingZeroes(false))
+      .withItems(new HridSetting().withStartNumber(9_999_999_999L).withRetainLeadingZeroes(false));
+
+    hridManager.updateHridSettings(newHridSettings)
+      .compose(v -> hridManager.getNextInstanceHrid())
+      .compose(hrid -> validateHrid(hrid, "9999999997", testContext))
+      .compose(v -> hridManager.getNextHoldingsHrid())
+      .compose(hrid -> validateHrid(hrid, "9999999998", testContext))
+      .compose(v -> hridManager.getNextItemHrid())
+      .compose(hrid -> validateHrid(hrid, "9999999999", testContext))
+      .onComplete(testContext.asyncAssertSuccess(
+        v1 -> log.info("Finished canGetNextHridWhenStartNumberIsLongWithoutLeadingZeroes()")));
+  }
+
```

Comment:
Why is this test needed? Why is it significantly different to `canGetNextHridWhenStartNumberIsLong` above it?

Reply:
I wrote these test also during investigation how HridManager works. These test also hasn't big sense.